pkg/releaseutil/manifest.go

```diff
@@ -40,7 +40,9 @@ func SplitManifests(bigfile string) map[string]string {
 	sep := "\n---\n"
 	tpl := "manifest-%d"
 	res := map[string]string{}
-	tmp := strings.Split(bigfile, sep)
+	// Making sure yaml formatting doesn't matter when generating manifest from string.
+	bigFileTmp := strings.TrimSpace(bigfile)
+	tmp := strings.Split(bigFileTmp, sep)
```

Comment:
- [ ] How about using [`(*regexp.Regexp).Split`](https://golang.org/pkg/regexp/#Regexp.Split) instead?  
  It looks like you want to match a YAML document separator with any amount of whitespace around it. Note that [a YAML document starting boundary marker](http://yaml.org/spec/current.html#c-document-start) must occur at the beginning of the line. Hence, I recommend the following regular expression:
```(?:^|\\s*\n)---\\s*```
That is, the boundary marker is either at the beginning of the document of occurring after a newline, but preceded by any amount of whitespace.  

  [Here's a sample using this technique in the Go playground.](https://play.golang.org/p/FsdylWkbWU)